mc/query/eval.go

```diff
@@ -0,0 +1,372 @@
+package query
+
+import (
+	"fmt"
+	pb "github.com/mediachain/concat/proto"
+	"strings"
+)
+
+// query evaluation: very primitive eval with only simple statements
+// will have to do until we have an index and we can compile to sql.
+func EvalQuery(query *Query, stmts []*pb.Statement) ([]interface{}, error) {
+	nsfilter := makeNamespaceFilter(query)
+
+	cfilter, err := makeCriteriaFilter(query)
+	if err != nil {
+		return nil, err
+	}
+
+	rs, err := makeResultSet(query)
+	if err != nil {
+		return nil, err
+	}
+
+	rs.begin(len(stmts))
+	for _, stmt := range stmts {
+		if nsfilter(stmt) && cfilter(stmt) {
```

Comment:
Right, I'll write a little compiler from MCQL Queries to SQL to go with the index.